(EXPERIMENTAL) Prototyping with to-device sender key distribution

hughns opened the pull request.

hughns committed:
Prototyping for to-device key distribution

Show participant ID and some encryption status message
Allow encryption system to be chosen at point of room creation
Send cryptoVersion platform data to Posthog
Send key distribution stats to posthog
Send encryption type for CallStarted and CallEnded events
Update js-sdk

hughns committed:
Prototyping for to-device key distribution

Show participant ID and some encryption status message
Allow encryption system to be chosen at point of room creation
Send cryptoVersion platform data to Posthog
Send key distribution stats to posthog
Send encryption type for CallStarted and CallEnded events
Update js-sdk

hughns committed:
Allow for null MatrixClient

hughns committed:
Allow for null MatrixClient

hughns committed:
Use branch of js-sdk

hughns committed:
Use branch of js-sdk

hughns committed:
Merge branch 'hughns/do-device-key-distribution' of https://github.com/element-hq/element-call into hughns/to-device-key-distribution

hughns committed:
Update yarn.lock

hughns committed:
Bump js-sdk

hughns committed:
Merge branch 'livekit' into hughns/do-device-key-distribution

hughns committed:
Bump js-sdk

hughns committed:
Upgrade js-sdk to get forward secrecy

hughns committed:
Bump js-sdk

hughns committed:
Bump js-sdk

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/element-hq/element-call?pullRequest=2572) <br/>All committers have signed the CLA.

hughns committed:
Merge branch 'livekit' into hughns/to-device-key-distribution

hughns committed:
Bump js-sdk

hughns committed:
Bump js-sdk

hughns committed:
Include the hostname of where EC is running in rageshakes (#2616)

hughns committed:
Give user feedback if rageshake submission fails

Otherwise it swallows errors like 50x

hughns committed:
Rageshake logging improvements

Capture MatrixRTC related prefixed/child loggers from js-sdk.
Add explicit prefix to livekit log entries.

hughns committed:
Merge branch 'livekit' into hughns/to-device-key-distribution

hughns committed:
Intercept matrix_sdk logging via console

hughns committed:
Bump js-sdk to fix embedded mode

hughns committed:
Merge branch 'livekit' into hughns/do-device-key-distribution

hughns committed:
Request widget permission to send encryption key to device messages

hughns committed:
Merge branch 'livekit' into hughns/to-device-key-distribution

toger5 commented:
I will close this since we have this mostly merged with the new encryption manager.
@hughns @BillCarsonFr does this sound right to you?
